Test_Engine/Query/FullMessage.cs

```diff
@@ -0,0 +1,101 @@
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using System;
+using System.Linq;
+using System.Collections.Generic;
+using System.ComponentModel;
+using BH.oM.Reflection.Attributes;
+using BH.oM.Base;
+using BH.oM.Test;
+using BH.oM.Test.Results;
+
+namespace BH.Engine.Test
+{
+    public static partial class Query
+    {
+        /***************************************************/
+        /**** Public Methods                            ****/
+        /***************************************************/
+
+        [Description("Returns a full concatenated message for the test results and its inner result to the target max depth, only giving messages worse or equal to the provided severity level.")]
+        [Input("result", "The TestResult to give the full message for.")]
+        [Input("maxDepth", "Maximum level of recursiveness for inner TestInformation.")]
+        [Input("minSeverity", "The minimum level of severity of the information to report back.")]
+        [Output("message", "Full message for the TestResult")]
+        public static string FullMessage(this TestResult result, int maxDepth = 3, TestStatus minSeverity = TestStatus.Pass)
+        {
+            //Return an empty string if the status is not at least as severe as the provided minimum.
+            if (!result.Status.IsEqualOrMoreSevere(minSeverity))
+                return "";
+
+            string message = result.Description + Environment.NewLine;
+            message += result.Message + Environment.NewLine;
+
+            //Filter out only info that passes the severity level
+            List<ITestInformation> innerInfo = result.Information.Where(x => x.Status.IsEqualOrMoreSevere(minSeverity)).ToList();
+
+            if (maxDepth > 0 && innerInfo.Count > 0)
+            {
+                message += "Inner results:" + Environment.NewLine;
+
+                foreach (ITestInformation info in innerInfo)
+                {
+                    string innerMessage = IFullMessage(info, maxDepth - 1, minSeverity);
+                    string[] arr = innerMessage.Split(new string[] { Environment.NewLine }, StringSplitOptions.RemoveEmptyEntries);
+                    foreach (string s in arr)
+                    {
+                        message += "\t" + s + Environment.NewLine;
+                    }
+
+                }
+            }
+
+            return message;
+        }
+
+        /***************************************************/
+        /**** Public Methods - Interface                ****/
+        /***************************************************/
+
+        public static string IFullMessage(this ITestInformation result, int maxDepth = 3, TestStatus minSeverity = TestStatus.Pass)
+        {
+            return FullMessage(result, maxDepth, minSeverity);
```

Comment:
There is no `as dynamic` here so what is the point of this method ?

Reply:
🤦 you are right. Did not spot it as it was giving the same results due to the `RunExtensionMethod` in the fallback method. Will fix.